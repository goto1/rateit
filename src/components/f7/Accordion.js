import React from "react";
import PropTypes from "prop-types";

const getComponent = name => ({ className, children }) => {
  const classNames = `${name} ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

export const AccordionItem = getComponent("accordion-item");

AccordionItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export const AccordionToggle = getComponent("accordion-item-toggle");

AccordionToggle.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export const AccordionContent = getComponent("accordion-item-content");

AccordionContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

const Accordion = getComponent("accordion-list");

Accordion.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Accordion;
