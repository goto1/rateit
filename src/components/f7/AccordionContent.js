import React from "react";
import PropTypes from "prop-types";

const AccordionContent = ({ className, children }) => {
  const classNames = `accordion-item-content ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionContent;
