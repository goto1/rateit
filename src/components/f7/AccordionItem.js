import React from "react";
import PropTypes from "prop-types";

const AccordionItem = ({ className, children }) => {
  const classNames = `accordion-item ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionItem;
