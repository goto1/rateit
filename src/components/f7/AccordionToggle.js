import React from "react";
import PropTypes from "prop-types";

const AccordionToggle = ({ className, children }) => {
  const classNames = `accordion-item-toggle ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionToggle.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionToggle;
