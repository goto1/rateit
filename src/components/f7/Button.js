import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import omitBy from "lodash/omitBy";
import isNil from "lodash/isNil";

const themePicker = (color, fill) => {
  const colors = {
    blue: "#057AFF",
    green: "#369947",
    red: "#C9302C",
    white: "#F1F1F5"
  };
  let theme = {
    background: "inherit",
    border: colors.blue,
    text: colors.blue
  };

  switch (color) {
    case "green":
      theme = fill
        ? { background: colors.green, border: colors.green, text: colors.white }
        : { ...theme, border: colors.green, text: colors.green };
      break;
    case "red":
      theme = fill
        ? {
            background: colors.red,
            border: colors.red,
            text: colors.white
          }
        : { ...theme, border: colors.red, text: colors.red };
      break;
    default:
      theme = fill
        ? {
            ...theme,
            background: colors.blue,
            text: colors.white
          }
        : theme;
  }
  return theme;
};

const Wrapper = styled.a`
  background-color: ${props => props.background} !important;
  border-color: ${props => props.border} !important;
  color: ${props => props.text} !important;
  opacity: ${props => (props.disabled ? ".65" : "1")};
  text-transform: uppercase;
  letter-spacing: .5px;
`;

const Button = ({
  big,
  fill,
  href,
  disabled,
  color,
  onClick,
  className,
  children
}) => {
  const { background, border, text } = themePicker(color, fill);
  const classNames = `button ${big ? "button-big" : ""} ${className
    ? className
    : ""}`.trim();
  const props = omitBy(
    {
      background,
      border,
      text,
      href,
      disabled,
      onClick,
      className: classNames
    },
    isNil
  );

  return (
    <Wrapper {...props}>
      {children}
    </Wrapper>
  );
};

Button.propTypes = {
  big: PropTypes.bool,
  fill: PropTypes.bool,
  href: PropTypes.string,
  disabled: PropTypes.bool,
  color: PropTypes.oneOf(["green", "red"]),
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Button;
