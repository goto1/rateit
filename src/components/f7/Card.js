import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`
  margin: ${props => (props.margin ? `${props.margin} !important` : "10px")};
`;

const Card = ({ margin, children }) =>
  <Wrapper margin={margin} className="card">
    {children}
  </Wrapper>;

Card.propTypes = {
  margin: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Card;
