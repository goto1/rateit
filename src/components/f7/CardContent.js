import React from "react";
import PropTypes from "prop-types";

const CardContent = ({ className, children }) => {
  const classNames = `card-content ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      <div className="card-content-inner">
        {children}
      </div>
    </div>
  );
};

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default CardContent;
