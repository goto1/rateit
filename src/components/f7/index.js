import Accordion, {
  AccordionContent,
  AccordionItem,
  AccordionToggle
} from "./Accordion";
import Button from "./Button";
import Card, { CardContent, CardFooter, CardHeader } from "./Card";
import ContentBlock from "./ContentBlock";
import ContentBlockTitle from "./ContentBlockTitle";
import FormInput from "./FormInput";
import Icon from "./Icon";
import InputElement from "./InputElement";
import LinkItem from "./LinkItem";
import ListBlock from "./ListBlock";
import ListItem from "./ListItem";
import Navbar, { NavLeft, NavCenter, NavRight } from "./Navbar";
import SmartSelect from "./SmartSelect";
import Tabs from "./Tabs";
import Textarea from "./Textarea";

export {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionToggle,
  Button,
  Card,
  CardContent,
  CardFooter,
  CardHeader,
  ContentBlock,
  ContentBlockTitle,
  FormInput,
  Icon,
  InputElement,
  LinkItem,
  ListBlock,
  ListItem,
  NavCenter,
  NavLeft,
  NavRight,
  Navbar,
  SmartSelect,
  Tabs,
  Textarea
};
