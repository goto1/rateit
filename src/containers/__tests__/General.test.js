import React from "react";
import {
  AccountInformation,
  SchoolInformation,
  PasswordReset,
  UserInformation
} from "../General";
import { Button, InputElement, SmartSelect } from "../../components/f7";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("General container", () => {
  describe("UserInformation", () => {
    const getForm = props => {
      const newProps = {
        majors: [
          { id: "hvdMFjM", name: "Computer Science", abbreviation: "CS" },
          { id: "e2LejDg", name: "Information Systems", abbreviation: "IS" }
        ],
        schools: [
          {
            id: "bruHy",
            name: "New Jersey Institute of Technology, Newark",
            abbreviation: "NJIT"
          },
          {
            id: "5y7fh",
            name: "Rutgers University, New Brunswick",
            abbreviation: "RU"
          }
        ],
        major: ["hvdMFjM", "e2LejDg"],
        school: ["bruHy", "5y7fh"],
        ...props
      };
      return <UserInformation {...newProps} />;
    };

    it("should render correctly", () => {
      const Form = getForm();
      const tree = renderer.create(Form).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render AccountInformation, SchoolInformation, and PasswordReset components", () => {
      const Form = getForm();
      const wrapper = mount(Form);
      const accInfoExists = wrapper.find(AccountInformation).exists();
      const schoolInfoExists = wrapper.find(SchoolInformation).exists();
      const passResetExists = wrapper.find(PasswordReset).exists();
      const actual = accInfoExists && schoolInfoExists && passResetExists;
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render an InputElement for current password information", () => {
      const Form = getForm();
      const wrapper = mount(Form);
      const actual = wrapper.find(InputElement).at(3).props().name;
      const expected = "current_password";
      expect(actual).toBe(expected);
    });

    it("should render a Button for submitting the form", () => {
      const Form = getForm();
      const wrapper = mount(Form);
      const actual = wrapper.find(Button).props().type;
      const expected = "submit";
      expect(actual).toBe(expected);
    });
  });
});
