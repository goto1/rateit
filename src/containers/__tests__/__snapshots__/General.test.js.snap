// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General container UserInformation should render correctly 1`] = `
.jwltLt {
  background-color: #369947 !important;
  border-color: #369947 !important;
  color: #F1F1F5 !important;
  opacity: .65;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  width: 100%;
}

.cYrpQX {
  margin-top: 25px !important;
}

.fYkctb {
  margin-left: 15px;
}

.fYkctb input {
  color: black;
}

.bqyzQp {
  position: relative;
}

<form
  onSubmit={[Function]}
>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      Account information
    </div>
    <div
      className="list-block inset"
    >
      <ul>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              email
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="example@gmail.com"
              type="email"
              value=""
            />
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      School information
    </div>
    <div
      className="list-block"
    >
      <ul>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a school..."
          >
            <select
              multiple={true}
              name="school"
              onBlur={undefined}
              onChange={[Function]}
              value={
                Array [
                  "bruHy",
                  "5y7fh",
                ]
              }
            >
              <option
                data-display-as="NJIT"
                value="bruHy"
              >
                New Jersey Institute of Technology, Newark
              </option>
              <option
                data-display-as="RU"
                value="5y7fh"
              >
                Rutgers University, New Brunswick
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  School
                </div>
                <div
                  className="item-after"
                >
                  NJIT, RU
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a major..."
          >
            <select
              multiple={true}
              name="major"
              onBlur={undefined}
              onChange={[Function]}
              value={
                Array [
                  "hvdMFjM",
                  "e2LejDg",
                ]
              }
            >
              <option
                data-display-as="CS"
                value="hvdMFjM"
              >
                Computer Science
              </option>
              <option
                data-display-as="IS"
                value="e2LejDg"
              >
                Information Systems
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  Major
                </div>
                <div
                  className="item-after"
                >
                  CS, IS
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      Password reset
    </div>
    <div
      className="list-block inset"
    >
      <ul>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              lock_outline
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="New Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              lock_outline
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="password_repeated"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="New Password Repeated"
              type="password"
              value=""
            />
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div
    className="content-block  "
  >
    <button
      className="button button-big sc-bdVaJa jwltLt"
      disabled={true}
      type="submit"
    >
      Submit Changes
    </button>
  </div>
</form>
`;
